plugins/PostProcessingPlugin/scripts/TimeLapse.py

```diff
@@ -0,0 +1,100 @@
+                {
+                    "label": "Pause length",
+                    "description": "How long to wait (in ms) after camera was triggered.",
+                    "type": "int",
+                    "default_value": 700,
+                    "minimum_value": 0,
+                    "unit": "ms"
+                },
+                "park_print_head":
+                {
+                    "label": "Park Print Head",
+                    "description": "Park the print head out of the way. Assumes absolute positioning.",
+                    "type": "bool",
+                    "default_value": true
+                },
+                "head_park_x":
+                {
+                    "label": "Park Print Head X",
+                    "description": "What X location does the head move to for photo.",
+                    "unit": "mm",
+                    "type": "float",
+                    "default_value": 0,
+                    "enabled": "park_print_head"
+                },
+                "head_park_y":
+                {
+                    "label": "Park Print Head Y",
+                    "description": "What Y location does the head move to for photo.",
+                    "unit": "mm",
+                    "type": "float",
+                    "default_value": 190,
+                    "enabled": "park_print_head"
+                },
+                "park_feed_rate":
+                {
+                    "label": "Park Feed Rate",
+                    "description": "How fast does the head move to the park coordinates.",
+                    "unit": "mm/s",
+                    "type": "float",
+                    "default_value": 9000,
+                    "enabled": "park_print_head"
+                }
+            }
+        }"""
+
+    def execute(self, data):
+        in_layer = False
+        feed_rate = self.getSettingValueByKey("park_feed_rate")
+        park_print_head = self.getSettingValueByKey("park_print_head")
+        x_park = self.getSettingValueByKey("head_park_x")
+        y_park = self.getSettingValueByKey("head_park_y")
+        trigger_cmd = self.getSettingValueByKey("trigger_cmd")
+        pause_length = self.getSettingValueByKey("pause_length")
+        gcode_to_append = ";TimeLapse Begin\n"
+
+        if park_print_head:
+            gcode_to_append += self.putValue(G = 1, F = feed_rate, X = x_park, Y = y_park) + ";Park print head\n"
+        gcode_to_append += self.putValue(M = 400) + ";Wait for moves to finish\n"
+        gcode_to_append += trigger_cmd + ";Snap Photo\n"
+        gcode_to_append += self.putValue(G = 4, P = pause_length) + ";Wait for camera\n"
+        gcode_to_append += ";TimeLapse End\n"
+        for layer in data:
+            # Check that a layer is being printed
+            lines = layer.split("\n")
+            if ";LAYER:" in lines[0]:
+                in_layer = True
+            else:
+                in_layer = False
+
+            if in_layer:
```

Comment:
Suggestion: This can be simplified by using `if ";LAYER:" in lines[0]:` and remove the 4 previous layers.